File: app/src/main/java/com/hqq/core/example/dialog/LeftDialog.java
package com.hqq.core.example.dialog;

import androidx.fragment.app.FragmentManager;
import android.view.Gravity;
import android.view.WindowManager;

import com.hqq.core.example.R;
import com.hqq.core.ui.BaseDialog;
import com.hqq.core.utils.ResourcesUtils;

/**
 * @version V1.0 <描述当前版本功能>
 * @author: huangqiqiang
 * 在此写用途
 * @FileName: com.hqq.core.app.dialog.RightDialog.java
 * @emain: 593979591@qq.com
 * @date: 2019-07-03 22:22
 */
public class LeftDialog extends BaseDialog {
    public static void showDialog(FragmentManager fragmentManager) {
        LeftDialog fragment = new LeftDialog();
        fragment.show(fragmentManager);
    }

    @Override
    public int setViewId() {
        return R.layout.dialog_right;
    }

    @Override
    public void initView() {

    }
    @Override
    public int setWeight() {
        return (int) ResourcesUtils.getDimen(getActivity(), R.dimen.x200);
    }

    @Override
    public int setHeight() {
        return WindowManager.LayoutParams.MATCH_PARENT;
    }

    @Override
    public int setGravity() {
        return Gravity.LEFT;
    }

    @Override
    public int setAnimation() {
        return R.anim.dialog_left_right;
    }
}

File: app/src/main/java/com/easy/example/dialog/LeftDialog.kt
package com.easy.example.dialog

import android.view.Gravity
import androidx.fragment.app.FragmentManager
import com.easy.core.ui.dialog.BaseDialog
import com.easy.example.R

/**
 * @version V1.0 <描述当前版本功能>
 * @author: huangqiqiang
 * 在此写用途
 * @FileName: com.easy.core.app.dialog.RightDialog.java
 * @emain: 593979591@qq.com
 * @date: 2019-07-03 22:22
</描述当前版本功能> */
class LeftDialog : BaseDialog() {
    companion object {
        fun showDialog(fragmentManager: FragmentManager?) {
            val fragment = LeftDialog()
            fragment.show(fragmentManager!!)
        }
    }

    override fun getGravity(): Int {
        return Gravity.LEFT
    }

    override fun getAnimation(): Int {
        return  R.anim.dialog_left_right
    }
    override fun getDialogLayoutId(): Int {
        return   R.layout.dialog_right
    }

    override fun initView() {}

}